### ISSUE TITLE
Generates empty CHANGELOG 

### ISSUE THREAD

**Issue by kaushalmodi** [orhun/git-cliff#75]
**Describe the bug**

I have tried all the different switches of `git cliff` but the output is always empty:

```html
# Changelog

All notable changes to this project will be documented in this file.

<!-- generated by git-cliff -->
```

**To Reproduce**

1. Download and extract `git-cliff` from https://github.com/orhun/git-cliff/releases/download/v0.6.1/git-cliff-0.6.1-x86_64-unknown-linux-musl.tar.gz and put it in `PATH`.
2. `git clone https://github.com/orhun/git-cliff`
3. `cd git-cliff`
4. `git cliff`
5. See this output
    ```html
    # Changelog

    All notable changes to this project will be documented in this file.

    <!-- generated by git-cliff -->
    ```

- I tried `git cliff -r .`, `git cliff -r . -w .`, `git cliff -l`, `git cliff -u`, `git cliff HEAD~5..`, but regardless, that output is the same as above.
- I also tried moving my `~/.gitconfig` temporarily thinking that something in there might be messing with this. But that didn't help either.

**Expected behavior**

I expected the git commit logs to be parsed by `git-cliff`.

**Screenshots/Logs**

![image](https://user-images.githubusercontent.com/3578197/161366421-ac0fff1b-3862-4836-8006-d90f1cc759b9.png)

**System (please complete the following information):**
 - OS Information: CentOS Linux 7.6.1810
 - Project Version: latest HEAD of this repo (0.6.1)

**Additional context**

- git version 2.34.0

**Comment by kaushalmodi** [comment]
I tried `git-cliff` using 3 install methods but I got the same outcome:

1. Download + extract the musl tar.gz from GitHub Releases
2. Clone this repo and `CARGO_TARGET_DIR=target cargo build --release`
3. `cargo install git-cliff`

I think that leaves out my OS and git versions as variables here.

**Comment by orhun** [comment]
Do you have `TEMPLATE` environment variable set to empty by any chance? It looks like `git-cliff` is using a blank template while generating the changelog.

Do any of the examples work? e.g. `git cliff -c examples/minimal` or `git cliff -c config/cliff.toml`  

Also, can you run `git-cliff` with `-vv` option and share the output?

**Comment by kaushalmodi** [comment]
> Do you have TEMPLATE environment variable set to empty by any chance?

No, it is undefined.

> Do any of the examples work? e.g. git cliff -c examples/minimal or git cliff -c config/cliff.toml

No, they give the same empty output.

> Also, can you run git-cliff with -vv option 

The debug info for `-vv` looks the same as `-v`.

> and share the output?

![image](https://user-images.githubusercontent.com/3578197/161392242-440e5dce-516c-47da-b461-4f074940a56d.png)

**Comment by kaushalmodi** [comment]
I noticed that the minimal example needed the .toml extension. With that, I get

![image](https://user-images.githubusercontent.com/3578197/161392285-5f4080d6-89c7-4870-9041-284fa1836cb6.png)

But still no changelog.

**Comment by kaushalmodi** [comment]
I also unset these env variables.. just in case.. `LESS`, `PAGER`, `GIT_PAGER`. That didn't help either.

**Comment by orhun** [comment]
I'm having trouble reproducing this issue. Can you also share the version of `libgit2`? Maybe it is related.

Have you tried running `git-cliff` with another repository? Do you get the same result?

**Comment by orhun** [comment]
Can you build `git-cliff` from source (`cargo build`) and share the output of `target/debug/git-cliff --context`?

**Comment by kaushalmodi** [comment]
Turns out that your latest commit https://github.com/orhun/git-cliff/commit/84507dd361f4c1ba37bfdb7005714bb817afb6b0 fixed this issue for me!

I had `INCLUDE_PATH` path set to point to some C header directory for GCC compilation. I am glad that you fixed above by prefixing `GIT_CLIFF_..`.

---

The issue is now fixed, but I will answer your questions anyways.

> Can you also share the version of libgit2? Maybe it is related.

I am using libgit2 that I built about 2 years back: https://github.com/libgit2/libgit2/commit/66137ff6e (v1.4.2+)

> Have you tried running git-cliff with another repository? Do you get the same result?

Yes. I was trying to run `git cliff` on your repo after it didn't work on any of mine 😄.

>Can you build git-cliff from source (cargo build) 

Now `cargo build --release` works as well.

---

Thanks!

**Comment by kaushalmodi** [comment]
@orhun May I suggest that when the user uses `-v` (or may be `-vv`), you print all the environment variables and parsed `git cliff` CLI switch values that affect its output?

**Comment by orhun** [comment]
Hey, I'm glad this issue is resolved by #76, wasn't expecting that 😄 

> @orhun May I suggest that when the user uses `-v` (or may be `-vv`), you print all the environment variables and parsed `git cliff` CLI switch values that affect its output?

Hmm, can you elaborate what you mean with an example? Actually, having a separate issue for tracking this would be great.
